### PR TITLE
Grace/hist page

### PR DESCRIPTION
**Pull request by gracewanggw**
# Year and Clerids  Changes on Historical Data Page

Make year selection a drop down instead of typing input. Remove clerid data from visualization

## Type of Change

- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update

## Checklist

- [ ] Unit/integration tests
- [ ] Documentation

## Tickets

- [#538 ](Link)
- [#537  ](Link)

## Screenshots

Please attach any design screenshots if UI update.

### CODE REVIEW

File: src/components/input-components/choice-input/index.js [hunk 1]
```diff
@@ -33,17 +36,42 @@ const ChoiceInput = (props) => {
...
+  )) : null;
+
   useEffect(() => {
     setFirstOptionText(value ? CLEAR_TEXT : initialFirstOptionText);
   }, [value, initialFirstOptionText]);
 
+  useEffect(() => {
+    if (isMulti) {
+      console.log('Multi updated');
+      console.log(vals);
```
Comment (tmonfre) [inline src/components/input-components/choice-input/index.js]: should remove these logs before merging

File: src/components/input-components/choice-input/index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /* eslint-disable jsx-a11y/label-has-for */
 /* eslint-disable jsx-a11y/label-has-associated-control */
 import React, { useState, useEffect } from 'react';
+import { MultiSelect } from 'react-multi-select-component';
```
Comment (tmonfre) [inline src/components/input-components/choice-input/index.js]: FYI I think you can just add `multiple` to the `<select>` tag to allow multi-select too. Not sure if you tried that out or not

Comment (tmonfre) [reply src/components/input-components/choice-input/index.js]: But using the external package is also fine 😄

File: src/constants/index.js [hunk 3]
```diff
@@ -17,6 +17,12 @@ const stateAbbrevToStateName = require('./state-abbreviations.json');
 const stateAbbrevToStateId = require('./state-ids.json');
 const stateAbbrevToZoomLevel = require('./state-zoom-levels.json');
 
+const yearsOptions = [
+  2022, 2021, 2020, 2019, 2018, 2017, 2016, 2015, 2014, 2013, 2012, 2011, 2010, 2009, 2008, 2007,
+  2006, 2005, 2004, 2003, 2002, 2001, 2000, 1999, 1998, 1997, 1996, 1995, 1994, 1993, 1992, 1991,
+  1990, 1989, 1988,
+];
```
Comment (tmonfre) [inline src/constants/index.js]: Hmm this won't account for when there are future years of data. There's logic in the selections reducer to figure out what the first and last years are when the data is fetched. Maybe you could store that in the reducer separately and use those values to generate this array in the component

File: src/screens/trapping-data/components/selection-bar/component.js [hunk 4]
```diff
@@ -39,16 +39,18 @@ const SelectionBar = (props) => {
   const selectedStateName = getStateNameFromAbbreviation(selectedState);
   const setStateAbbrev = stateName => setState(getStateAbbreviationFromStateName(stateName));
 
+  console.log(statesMappedToNames);
```
Comment (tmonfre) [inline src/screens/trapping-data/components/selection-bar/component.js]: Should remove before merging

File: src/components/input-components/choice-input/index.js (hunk 2 above)
Comment (AlejandroALopez) [reply src/components/input-components/choice-input/index.js]: Hi Thomas! We are making a personalized multi-selection component based on the figma designs. I think that this MultiSelect component might be from a separate branch I made and got here, but I am planning to replace it in my next PR

File: src/screens/trapping-data/components/selection-bar/component.js [hunk 5]
```diff
@@ -25,10 +25,20 @@ const SelectionBar = (props) => {
     setRangerDistrict,
     setStartYear,
     setState,
-    startYear,
     trappingData,
   } = props;
 
+  function generateYearOptions() {
```
Comment (AlejandroALopez) [inline src/screens/trapping-data/components/selection-bar/component.js]: We would want a syntax like `const yearOptions = () => { all the content here }`. We normally don't want to call them functions

File: src/screens/trapping-data/components/selection-bar/component.js [hunk 6]
```diff
@@ -39,16 +49,17 @@ const SelectionBar = (props) => {
...
-        <div id="start-year-selection"><TextInput instructions="Year" setValue={setStartYear} value={startYear} /></div>
+        <div id="start-year-selection"><ChoiceInput instructions="Year" setValue={setStartYear} options={generateYearOptions()} /></div>
         <div id="vl3" />
         {/* TODO: "to" */}
-        <div id="end-year-selection"><TextInput setValue={setEndYear} value={endYear} /></div>
+        <div id="end-year-selection"><ChoiceInput setValue={setEndYear} options={generateYearOptions()} /></div>
       </div>
       <div id="vl1" />
-      <ChoiceInput instructions="Select State" value={selectedStateName} setValue={setStateAbbrev} options={statesMappedToNames} firstOptionText="State" />
+      <ChoiceInput instructions="Select State" value={selectedStateName} setValue={setStateAbbrev} options={statesMappedToNames} firstOptionText="State" isMulti={false} />
```
Comment (AlejandroALopez) [inline src/screens/trapping-data/components/selection-bar/component.js]: remove the `is Multi` thing

File: src/screens/trapping-data/components/selection-bar/component.js [hunk 7]
```diff
@@ -77,6 +88,7 @@ const SelectionBar = (props) => {
             setValue={countyMode ? setCounty : setRangerDistrict}
             options={countyMode ? allCounties : allRangerDistricts}
             firstOptionText={countyMode ? 'County' : 'Ranger District'}
+            isMulti
```
Comment (AlejandroALopez) [inline src/screens/trapping-data/components/selection-bar/component.js]: here too

**Review by AlejandroALopez — APPROVED** [review]
Good job! We might replace the for loop later if we find a better alternative, but it might be good for now.

**Pull request merged**